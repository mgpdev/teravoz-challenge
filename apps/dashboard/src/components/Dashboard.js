import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import Fade from "@material-ui/core/Fade";
import Paper from "@material-ui/core/Paper";
import CallList from "./CallList";
import FinishedCallList from "./FinishedCallList";

const styles = theme => ({
  root: {
    flexGrow: 1,
    padding: 40
  },
  paper: {
    ...theme.mixins.gutters(),
    paddingTop: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit * 2
  }
});

function Dashboard(props) {
  const { classes } = props;
  return (
    <div className={classes.root}>
      <Fade in={true}>
        <Grid container spacing={24}>
          <Grid item xs={12} md={6}>
            <Paper className={classes.paper} elevation={1}>
              <CallList title="Active calls of new customers" queue="900" />
            </Paper>
          </Grid>
          <Grid item xs={12} md={6}>
            <Paper className={classes.paper} elevation={1}>
              <CallList
                title="Active calls of returning customers"
                queue="901"
              />
            </Paper>
          </Grid>
          <Grid item xs={12}>
            <FinishedCallList />
          </Grid>
        </Grid>
      </Fade>
    </div>
  );
}

Dashboard.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(Dashboard);
