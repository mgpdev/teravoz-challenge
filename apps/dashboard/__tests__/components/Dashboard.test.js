import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

// Component to be tested
import Dashboard from "./../../src/components/Dashboard";

describe("<Dashboard />", () => {
  describe("render()", () => {
    test("renders the component", () => {
      const wrapper = shallow(<Dashboard />);

      const component = wrapper.html();

      expect(component).toMatchSnapshot();
    });
  });
});
