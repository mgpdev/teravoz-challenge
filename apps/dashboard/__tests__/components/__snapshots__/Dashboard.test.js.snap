// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> render() renders the component 1`] = `"<div class=\\"Dashboard-root-1\\"><div class=\\"MuiGrid-container-3 MuiGrid-spacing-xs-24-27\\" style=\\"opacity:0;will-change:opacity\\"><div class=\\"MuiGrid-item-4 MuiGrid-grid-xs-12-43 MuiGrid-grid-md-6-65\\"><div class=\\"MuiPaper-root-100 MuiPaper-elevation1-103 MuiPaper-rounded-101 Dashboard-paper-2\\"><div class=\\"CallList-root-127\\"><h2 class=\\"MuiTypography-root-128 MuiTypography-title-134 MuiTypography-gutterBottom-145\\">New customers calls</h2><ul class=\\"MuiList-root-153 MuiList-padding-154\\"><div><li class=\\"MuiListItem-root-157 MuiListItem-default-160 MuiListItem-gutters-164\\"><div class=\\"MuiAvatar-root-168 MuiAvatar-colorDefault-169\\"><svg class=\\"MuiSvgIcon-root-171\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path fill=\\"none\\" d=\\"M0 0h24v24H0V0z\\"></path><g><path d=\\"M19.23 15.26l-2.54-.29c-.61-.07-1.21.14-1.64.57l-1.84 1.84c-2.83-1.44-5.15-3.75-6.59-6.59l1.85-1.85c.43-.43.64-1.03.57-1.64l-.29-2.52c-.12-1.01-.97-1.77-1.99-1.77H5.03c-1.13 0-2.07.94-2 2.07.53 8.54 7.36 15.36 15.89 15.89 1.13.07 2.07-.87 2.07-2v-1.73c.01-1.01-.75-1.86-1.76-1.98z\\"></path></g></svg></div><div class=\\"MuiListItemText-root-178\\"><span class=\\"MuiTypography-root-128 MuiTypography-subheading-135 MuiListItemText-primary-181\\">(11) 9986-5645</span><p class=\\"MuiTypography-root-128 MuiTypography-body1-137 MuiTypography-colorTextSecondary-151 MuiListItemText-secondary-182\\">11:50</p></div></li></div><div><li class=\\"MuiListItem-root-157 MuiListItem-default-160 MuiListItem-gutters-164\\"><div class=\\"MuiAvatar-root-168 MuiAvatar-colorDefault-169\\"><svg class=\\"MuiSvgIcon-root-171\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path fill=\\"none\\" d=\\"M0 0h24v24H0V0z\\"></path><g><path d=\\"M19.23 15.26l-2.54-.29c-.61-.07-1.21.14-1.64.57l-1.84 1.84c-2.83-1.44-5.15-3.75-6.59-6.59l1.85-1.85c.43-.43.64-1.03.57-1.64l-.29-2.52c-.12-1.01-.97-1.77-1.99-1.77H5.03c-1.13 0-2.07.94-2 2.07.53 8.54 7.36 15.36 15.89 15.89 1.13.07 2.07-.87 2.07-2v-1.73c.01-1.01-.75-1.86-1.76-1.98z\\"></path></g></svg></div><div class=\\"MuiListItemText-root-178\\"><span class=\\"MuiTypography-root-128 MuiTypography-subheading-135 MuiListItemText-primary-181\\">(11) 9986-5645</span><p class=\\"MuiTypography-root-128 MuiTypography-body1-137 MuiTypography-colorTextSecondary-151 MuiListItemText-secondary-182\\">11:50</p></div></li></div><div><li class=\\"MuiListItem-root-157 MuiListItem-default-160 MuiListItem-gutters-164\\"><div class=\\"MuiAvatar-root-168 MuiAvatar-colorDefault-169\\"><svg class=\\"MuiSvgIcon-root-171\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path fill=\\"none\\" d=\\"M0 0h24v24H0V0z\\"></path><g><path d=\\"M19.23 15.26l-2.54-.29c-.61-.07-1.21.14-1.64.57l-1.84 1.84c-2.83-1.44-5.15-3.75-6.59-6.59l1.85-1.85c.43-.43.64-1.03.57-1.64l-.29-2.52c-.12-1.01-.97-1.77-1.99-1.77H5.03c-1.13 0-2.07.94-2 2.07.53 8.54 7.36 15.36 15.89 15.89 1.13.07 2.07-.87 2.07-2v-1.73c.01-1.01-.75-1.86-1.76-1.98z\\"></path></g></svg></div><div class=\\"MuiListItemText-root-178\\"><span class=\\"MuiTypography-root-128 MuiTypography-subheading-135 MuiListItemText-primary-181\\">(11) 9986-5645</span><p class=\\"MuiTypography-root-128 MuiTypography-body1-137 MuiTypography-colorTextSecondary-151 MuiListItemText-secondary-182\\">11:50</p></div></li></div></ul></div></div></div><div class=\\"MuiGrid-item-4 MuiGrid-grid-xs-12-43 MuiGrid-grid-md-6-65\\"><div class=\\"MuiPaper-root-100 MuiPaper-elevation1-103 MuiPaper-rounded-101 Dashboard-paper-2\\"><div class=\\"CallList-root-127\\"><h2 class=\\"MuiTypography-root-128 MuiTypography-title-134 MuiTypography-gutterBottom-145\\">Returning customers calls</h2><ul class=\\"MuiList-root-153 MuiList-padding-154\\"><div><li class=\\"MuiListItem-root-157 MuiListItem-default-160 MuiListItem-gutters-164\\"><div class=\\"MuiAvatar-root-168 MuiAvatar-colorDefault-169\\"><svg class=\\"MuiSvgIcon-root-171\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path fill=\\"none\\" d=\\"M0 0h24v24H0V0z\\"></path><g><path d=\\"M19.23 15.26l-2.54-.29c-.61-.07-1.21.14-1.64.57l-1.84 1.84c-2.83-1.44-5.15-3.75-6.59-6.59l1.85-1.85c.43-.43.64-1.03.57-1.64l-.29-2.52c-.12-1.01-.97-1.77-1.99-1.77H5.03c-1.13 0-2.07.94-2 2.07.53 8.54 7.36 15.36 15.89 15.89 1.13.07 2.07-.87 2.07-2v-1.73c.01-1.01-.75-1.86-1.76-1.98z\\"></path></g></svg></div><div class=\\"MuiListItemText-root-178\\"><span class=\\"MuiTypography-root-128 MuiTypography-subheading-135 MuiListItemText-primary-181\\">(11) 9986-5645</span><p class=\\"MuiTypography-root-128 MuiTypography-body1-137 MuiTypography-colorTextSecondary-151 MuiListItemText-secondary-182\\">11:50</p></div></li></div><div><li class=\\"MuiListItem-root-157 MuiListItem-default-160 MuiListItem-gutters-164\\"><div class=\\"MuiAvatar-root-168 MuiAvatar-colorDefault-169\\"><svg class=\\"MuiSvgIcon-root-171\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path fill=\\"none\\" d=\\"M0 0h24v24H0V0z\\"></path><g><path d=\\"M19.23 15.26l-2.54-.29c-.61-.07-1.21.14-1.64.57l-1.84 1.84c-2.83-1.44-5.15-3.75-6.59-6.59l1.85-1.85c.43-.43.64-1.03.57-1.64l-.29-2.52c-.12-1.01-.97-1.77-1.99-1.77H5.03c-1.13 0-2.07.94-2 2.07.53 8.54 7.36 15.36 15.89 15.89 1.13.07 2.07-.87 2.07-2v-1.73c.01-1.01-.75-1.86-1.76-1.98z\\"></path></g></svg></div><div class=\\"MuiListItemText-root-178\\"><span class=\\"MuiTypography-root-128 MuiTypography-subheading-135 MuiListItemText-primary-181\\">(11) 9986-5645</span><p class=\\"MuiTypography-root-128 MuiTypography-body1-137 MuiTypography-colorTextSecondary-151 MuiListItemText-secondary-182\\">11:50</p></div></li></div><div><li class=\\"MuiListItem-root-157 MuiListItem-default-160 MuiListItem-gutters-164\\"><div class=\\"MuiAvatar-root-168 MuiAvatar-colorDefault-169\\"><svg class=\\"MuiSvgIcon-root-171\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path fill=\\"none\\" d=\\"M0 0h24v24H0V0z\\"></path><g><path d=\\"M19.23 15.26l-2.54-.29c-.61-.07-1.21.14-1.64.57l-1.84 1.84c-2.83-1.44-5.15-3.75-6.59-6.59l1.85-1.85c.43-.43.64-1.03.57-1.64l-.29-2.52c-.12-1.01-.97-1.77-1.99-1.77H5.03c-1.13 0-2.07.94-2 2.07.53 8.54 7.36 15.36 15.89 15.89 1.13.07 2.07-.87 2.07-2v-1.73c.01-1.01-.75-1.86-1.76-1.98z\\"></path></g></svg></div><div class=\\"MuiListItemText-root-178\\"><span class=\\"MuiTypography-root-128 MuiTypography-subheading-135 MuiListItemText-primary-181\\">(11) 9986-5645</span><p class=\\"MuiTypography-root-128 MuiTypography-body1-137 MuiTypography-colorTextSecondary-151 MuiListItemText-secondary-182\\">11:50</p></div></li></div></ul></div></div></div><div class=\\"MuiGrid-item-4 MuiGrid-grid-xs-12-43\\"><div class=\\"MuiPaper-root-100 MuiPaper-elevation2-104 MuiPaper-rounded-101 FinishedCallList-root-184\\"><h2 class=\\"MuiTypography-root-128 MuiTypography-title-134 MuiTypography-gutterBottom-145 FinishedCallList-title-186\\">Finished Calls</h2><table class=\\"MuiTable-root-187 FinishedCallList-table-185\\"><thead class=\\"MuiTableHead-root-188\\"><tr class=\\"MuiTableRow-root-189 MuiTableRow-head-192\\"><th class=\\"MuiTableCell-root-194 MuiTableCell-head-195\\" scope=\\"col\\">Customer</th><th class=\\"MuiTableCell-root-194 MuiTableCell-head-195 MuiTableCell-numeric-198\\" scope=\\"col\\">Number</th><th class=\\"MuiTableCell-root-194 MuiTableCell-head-195 MuiTableCell-numeric-198\\" scope=\\"col\\">Start Time</th><th class=\\"MuiTableCell-root-194 MuiTableCell-head-195 MuiTableCell-numeric-198\\" scope=\\"col\\">End Time</th><th class=\\"MuiTableCell-root-194 MuiTableCell-head-195 MuiTableCell-numeric-198\\" scope=\\"col\\">Download</th></tr></thead><tbody class=\\"MuiTableBody-root-202\\"><tr class=\\"MuiTableRow-root-189\\"><th class=\\"MuiTableCell-root-194 MuiTableCell-body-196\\" scope=\\"row\\">Frozen yoghurt</th><td class=\\"MuiTableCell-root-194 MuiTableCell-body-196 MuiTableCell-numeric-198\\">159</td><td class=\\"MuiTableCell-root-194 MuiTableCell-body-196 MuiTableCell-numeric-198\\">6</td><td class=\\"MuiTableCell-root-194 MuiTableCell-body-196 MuiTableCell-numeric-198\\">24</td><td class=\\"MuiTableCell-root-194 MuiTableCell-body-196 MuiTableCell-numeric-198\\">4</td></tr><tr class=\\"MuiTableRow-root-189\\"><th class=\\"MuiTableCell-root-194 MuiTableCell-body-196\\" scope=\\"row\\">Ice cream sandwich</th><td class=\\"MuiTableCell-root-194 MuiTableCell-body-196 MuiTableCell-numeric-198\\">237</td><td class=\\"MuiTableCell-root-194 MuiTableCell-body-196 MuiTableCell-numeric-198\\">9</td><td class=\\"MuiTableCell-root-194 MuiTableCell-body-196 MuiTableCell-numeric-198\\">37</td><td class=\\"MuiTableCell-root-194 MuiTableCell-body-196 MuiTableCell-numeric-198\\">4.3</td></tr><tr class=\\"MuiTableRow-root-189\\"><th class=\\"MuiTableCell-root-194 MuiTableCell-body-196\\" scope=\\"row\\">Eclair</th><td class=\\"MuiTableCell-root-194 MuiTableCell-body-196 MuiTableCell-numeric-198\\">262</td><td class=\\"MuiTableCell-root-194 MuiTableCell-body-196 MuiTableCell-numeric-198\\">16</td><td class=\\"MuiTableCell-root-194 MuiTableCell-body-196 MuiTableCell-numeric-198\\">24</td><td class=\\"MuiTableCell-root-194 MuiTableCell-body-196 MuiTableCell-numeric-198\\">6</td></tr><tr class=\\"MuiTableRow-root-189\\"><th class=\\"MuiTableCell-root-194 MuiTableCell-body-196\\" scope=\\"row\\">Cupcake</th><td class=\\"MuiTableCell-root-194 MuiTableCell-body-196 MuiTableCell-numeric-198\\">305</td><td class=\\"MuiTableCell-root-194 MuiTableCell-body-196 MuiTableCell-numeric-198\\">3.7</td><td class=\\"MuiTableCell-root-194 MuiTableCell-body-196 MuiTableCell-numeric-198\\">67</td><td class=\\"MuiTableCell-root-194 MuiTableCell-body-196 MuiTableCell-numeric-198\\">4.3</td></tr><tr class=\\"MuiTableRow-root-189\\"><th class=\\"MuiTableCell-root-194 MuiTableCell-body-196\\" scope=\\"row\\">Gingerbread</th><td class=\\"MuiTableCell-root-194 MuiTableCell-body-196 MuiTableCell-numeric-198\\">356</td><td class=\\"MuiTableCell-root-194 MuiTableCell-body-196 MuiTableCell-numeric-198\\">16</td><td class=\\"MuiTableCell-root-194 MuiTableCell-body-196 MuiTableCell-numeric-198\\">49</td><td class=\\"MuiTableCell-root-194 MuiTableCell-body-196 MuiTableCell-numeric-198\\">3.9</td></tr></tbody></table></div></div></div></div>"`;
